Terminal.Gui/Core/MainLoop.cs

```diff
@@ -45,21 +45,46 @@ public interface IMainLoopDriver {
 	///   does not seem to be a way of supporting this on Windows.
 	/// </remarks>
 	public class MainLoop {
-		internal class Timeout {
+		/// <summary>
+		/// Provides data for timers running manipulation.
+		/// </summary>
+		public sealed class Timeout {
+			/// <summary>
+			/// Time to wait before invoke the callback.
+			/// </summary>
 			public TimeSpan Span;
+			/// <summary>
+			/// The function that will be invoked.
+			/// </summary>
 			public Func<MainLoop, bool> Callback;
 		}
 
 		internal SortedList<long, Timeout> timeouts = new SortedList<long, Timeout> ();
 		object timeoutsLockToken = new object ();
 		internal List<Func<bool>> idleHandlers = new List<Func<bool>> ();
 
+		/// <summary>
+		/// Gets the list of all timeouts.
+		/// </summary>
+		public SortedList<long, Timeout> Timeouts => timeouts;
```

Comment:
@tig is this enough?

```cs
		/// <summary>
		/// Gets the list of all timeouts sorted by the <see cref="TimeSpan"/> time ticks./>.
		/// A shorter limit time can be added at the end, but it will be called before an
		///  earlier addition that has a longer limit.
		/// </summary>
		public SortedList<long, Timeout> Timeouts => timeouts;
```